### ISSUE TITLE
Sound crackle problem

### ISSUE THREAD

**Issue by MehdiRaash** [CasparCG/server#1330]
> please watch the [ issue ](https://youtu.be/zejvYhvDwSY)

In our case, we have been facing a kind of "crackle" sound problem.

**It happens when**: about 1 hour or so **after** connecting to the *reference* the normal playing sound becomes weak and turns to crackling kind of sound.

In case we run this on Free Run, every thing works perfectly fine!

**Detailed configuration**:

* Refrence: Black Burst
* Card: Blackmagic quad 2
* OS: windows 10
* GPU: quadro P4000
* CPU: 2X GOLD 5118
* CasparCG version: 2.3.0 LTS RC
* SDK version: BM 10.8.2

### Environment
* Commit: [e.g. ab1234c]
* Server version: [e.g. v2.2]
* Operating system: [e.g. Windows 10]

---
I recorded a [video](https://youtu.be/zejvYhvDwSY) while crackling happened,

**Comment by unxoun** [comment]
Oh God I got the same problem!

**Comment by saltomodules** [comment]
Updating to newer decklink drivers might help. 

We had similar crackling issues with 10.x.x drivers, albeit on Ubuntu 18.04.

After updating to 11.3+ drivers the issue seems to be gone. Even when running a decklink input for several days.

**Comment by 5opr4ni** [comment]
Hi

I think that BM requires Trilevel not Blackburst for HD.

/olle

**Comment by dotarmin** [comment]
> I think that BM requires Trilevel not Blackburst for HD.

That's correct. Decklink drivers version >= 10.9.x seems to require Trilevel for HD while it works fine with version <= 10.8.x.

**Comment by MehdiRaash** [comment]
@dotarmin @5opr4ni @saltomodules 
- Firstly, we are on **SD** not HD, and the current version of blackMagic is **10.8.2.**
- Secondly, we have tested all the versions above **blackMagic 11.x.x**.

**The problem hasn't been solved yet, regarding the above sentences**

_Moreover, it's about 26 hours or so we have been running casparCG 2.2.0 stable, with the same configuration, **up to now** everything is Okay._

**Comment by dotarmin** [comment]
Hi @MehdiRaash 

> Firstly, we are on SD not HD, and the current version of blackMagic is 10.8.2.

We are just trying to help you out based on some issues that we have stumbled across. You have not written anywhere that you are on SD and not HD so the tips are based on what we usein our environment and on our experience.  

Try to post some logs, a configuration file and even maybe a screenshot from DIAG window and somebody from the community can spot what's wrong. 

I noticed you wrote that you use 2X GOLD 5118 for CPU, I'm not sure but I think somebody in the community told me that support for dual CPUs is not working that good in the server as is...(??) @ronag or @Julusian, can it possibly be someone of you that mentioned that a long time ago?

**Comment by saltomodules** [comment]
In our case (on Ubuntu 18.04) it's a 2.3 build from commit 45ebbf1f2 (17 march 2020) that is running without crackling.

Perhaps you could test a windows build near that one (the exact commit is not available on the build server), for instance:
http://casparcg.com/builds/CasparCG%20Server/master/casparcg-server-2b8a42a125ceaa81faa8458659af46b3f15cfe33-windows.zip

If that build also does run without crackling on Windows, perhaps that could help isolating the issue.

**Comment by MehdiRaash** [comment]
@dotarmin dear Dotarmin, there was no offense, language barrier I guess :-), English are both non-native for us.
I am so thankful for the platform you are developing.

**Comment by Rancbar** [comment]
Does any body have any suggestion to solve the problem?

This problem occurs when the Diagnostics window shows DeckLink card has late-frame issue. In the same time the Buffered-Audio goes to minimum level (and cannot raise up when the reference generator is connected.) (My two cents)

It seems the problem relates to buffered audio.

As the @dotarmin noticed about incompatibility may happens on Dual CPU servers we had repeated the test on a server with other specs too and late-frame problem is still there.

- CPU: Core I9-10920X 

- MotherBoard: PRIME X299-DELUXE II

- Card: DeckLink 8K Pro

### The bellow details is without reference signal and in free running mode.

[**Sample video of problem** (on youtube)](https://youtu.be/lDuiMP9KMYY)

**Some screen shots of CasparCG Diagnose screen when problem occurs:**

![error](https://user-images.githubusercontent.com/10092307/93459045-634f3500-f8f6-11ea-8b31-d2a05a2a17fe.png)

![error3](https://user-images.githubusercontent.com/10092307/93459055-677b5280-f8f6-11ea-81ad-0c24e5a2fea9.png)

**CasparCG logs**
  _I should mention that we added a log to CasparCG code when the **late-frame** occurs and compiled it to track the problem with high accuracy_

Sample: `[2020-09-17 00:24:30.259] [error]   DeckLink 8K Pro [1-3|1080i5000] late-frame`

[caspar_2020-09-17.log](https://github.com/CasparCG/server/files/5238447/caspar_2020-09-17.log)

**Comment by premultiply** [comment]
Sounds to me like an "simple" problem with different clocks and missing buffering/resampling for compensation.

To me its still not clear who is the timing master in CasparCG?
How is reference input handled?
How are different time reference clocks between channels or input and output are handled?

**Comment by Rancbar** [comment]
If you are speaking about the reference signal generator device, we have disconnected it and run CasparCG in free running mode.
And also the problem occurs in both of playing file or DeckLink device's input.

If it helps we wrote a simple service to synchronize all servers time (windows clock) every 30 seconds. Can it be the source of problem?

**Comment by TuomasJaa** [comment]
Any fixes?

I have the same problem.
1920i50 works, but in 1920P25 or 1920P50 audio slows down and there is jitter sound in SDI signal.
And from system sound the sound cuts of.
Decklink duo 2 card with 11.7 drivers.

**Comment by TomKaltz** [comment]
@Julusian do you suspect this is still an issue these days?  Anyone still having this issue with latest in 3.4+?
